### ISSUE TITLE
issue trying to recognize an unknown person

### ISSUE THREAD

**Issue by linengmiao** [cmusatyalab/openface#309]
Hi

I don't have many years of experiences when it comes to machine learning so was hoping some people here could tell me what I may be doing wrong. I have tried multiple different approaches, the one I explain below gave the best results, but still...

#The situation:

I am trying to implement a login system using face recognition. My system contains 5 authorized users and any other person showing their face, who are not part of the autorized users, should be classified as unauthorized and therefor unable to login.
My system contains: Elon Musk, Jason Statham, Angelina Jolie, Bill Gates  [and this user which we will call John](https://imgur.com/a/4iXFF) (this link contains all the pictures I use for this person)

I am using [this framework](https://cmusatyalab.github.io/openface/), which allows me to align every face using facial landmarks and afterwards using a pretrained deep neural network to extract 128 features of the face it sees in order to compare it with the features of the faces which are allowed.
For the actual comparison I implemented a one-vs-rest architecture of non-linear SVM's (with an rbf kernel) in python.
A code snippet showing how I set up the classifier:

    fname = "/home/technicalQuestion22/openface/demos/featuresDir/labels.csv"
    labels = pd.read_csv(fname, header=None).as_matrix()[:, 1]
    labels = map(itemgetter(1),
                 map(os.path.split,
                     map(os.path.dirname, labels)))  # Get the directory.
 
    fname = "/home/technicalQuestion22/openface/demos/featuresDir/reps.csv" #contains 128 datapoints for every valid picture
    embeddings = pd.read_csv(fname, header=None).as_matrix()
    le = LabelEncoder().fit(labels)
    labelsNum = le.transform(labels)
    nClasses = len(le.classes_)
 
    svcClassifier = SVC(kernel='rbf', probability=True)
    classifier = OneVsRestClassifier(svcClassifier).fit(embeddings, labelsNum)

Every authorized user has 12 pictures. From each pictures those 128 features are extracted.

Now, all of this works perfectly fine: when I show a new picture of eg Elon Musk it is correctly recognized with an average certitude of 87%, same with all the other user. When I show a picture of a face that the system has never seen before and which therefor should be unauthorized, the system classifies it eg as Bill gates with a certitude 2% (highest certitude).

#The issue:

Everything seems to work fine so far... except when I show this unauthorized person's face (see 2 pictures below). Let's call him criminal Michael:

For some dark unknown reason, Michael is being classified as John with a certitude of up to 82%!
The certitude fluctuates a lot, jumping from 20% straight to 82%, back to 55, etc...

79% certitude as being John: https://imgur.com/a/9zLIP

85%  certitude as being John: https://imgur.com/a/Fn0yx

#My questions:

1)  Why is Michael classified as being John with such a high certitude?

2) What can I do so that Michael is not classified as being John with such a high certitude?

#Remark:

It is not possible for me to take 1000 picture of every user.

Thank you very much for your input
